src/common/cockpitwebserver.c

```diff
@@ -1029,10 +1029,15 @@ on_request_input (GObject *pollable_input,
   gssize count;
 
   length = request->buffer->len;
-  g_byte_array_set_size (request->buffer, length + 4096);
+  /* Read all pending data; with some Gio versions and TLS this is not
+   * level-triggered, i. e. if there is more data available, the callback does
```

Comment:
please remove this language about edge- vs level- triggered.  The real issue is that we're not checking for pending data in the gnutls buffer.  But even then, I'd only mention that in the commit message and leave the code comment to just explain what we're trying to do here: read one more than the maximum request size to find out if we should hang up immediately or not.

Reply:
fixed